q2_annotate/kaiju/database.py

```diff
@@ -98,6 +103,109 @@ def _find_latest_db_url(response: bytes, database_type: str) -> str:
     raise ValueError(f"URL for database type '{database_type}' not found.")
 
 
+def _build_kaiju_bwt_index(fasta_file: str, output_dir: str, threads: int):
+    """
+    Build BWT index using kaiju-mkbwt.
+    
+    Args:
+        fasta_file: Path to input FASTA file
+        output_dir: Directory where index files will be created
+        threads: Number of threads to use
+    """
+    bwt_file = os.path.join(output_dir, "kaiju_db_idx.bwt")
+    sa_file = os.path.join(output_dir, "kaiju_db_idx.sa")
+    
+    cmd = [
+        "kaiju-mkbwt",
+        "-n", str(threads),
+        "-a", "ACDEFGHIKLMNPQRSTVWY",
+        "-o", bwt_file,
+        fasta_file
+    ]
+    
+    try:
+        run_command(cmd=cmd, verbose=True)
+    except subprocess.CalledProcessError as e:
+        raise Exception(
+            f"An error was encountered while building the BWT index, "
+            f"(return code {e.returncode}), please inspect "
+            "stdout and stderr to learn more."
+        )
+
+
+def _build_kaiju_fmi_index(output_dir: str):
+    """
+    Build FM index using kaiju-mkfmi.
+    
+    Args:
+        output_dir: Directory containing BWT files and where FM index will be created
+    """
+    bwt_file = os.path.join(output_dir, "kaiju_db_idx.bwt")
+    fmi_file = os.path.join(output_dir, "kaiju_db_idx.fmi")
+    
+    cmd = [
+        "kaiju-mkfmi",
+        bwt_file
+    ]
+    
+    try:
+        run_command(cmd=cmd, verbose=True)
+    except subprocess.CalledProcessError as e:
+        raise Exception(
+            f"An error was encountered while building the FM index, "
+            f"(return code {e.returncode}), please inspect "
+            "stdout and stderr to learn more."
+        )
+
+
+def build_kaiju_db(
+    seqs: DNAFASTAFormat,
```

Comment:
We actually should accept here a ProteinFASTAFormat as the `kaiju-mkbwt` command expects protein sequences.